assets/js/script.js

```diff
@@ -10,39 +16,50 @@ const currentPos = {
-	const footerHeight = document.querySelector('header').offsetHeight;
-	canvas.height = window.innerHeight -headerHeight -footerHeight;
+	canvas.height = window.innerHeight -headerHeight -footerHeight - 9;
 	canvas.width = window.innerWidth;
 }
-window.addEventListener('load', setDisplay);
-window.addEventListener('resize', setDisplay);
+
+//Dividing the canvas into pseudo grids.
+// Testing incrementing x and y pos of apple in increments of 30;
+
+// Number of X positions possible at intervals of 30: (0,30,60,90,.....)
+const xGridPositions = () => {
+  if (canvas.width / canvasElementsDim !== Math.floor(canvas.width / canvasElementsDim)) {    // Checking if the canvas width is exactly divisible by 30
+    xPositions = Math.floor(canvas.width / canvasElementsDim) - 1;             // If not then 30px box can't fit without overflow, we remove 1 position to make sure the box fits even if there is extra space at the right and bottom sides.
+                                                                // Ex: canvas width = 1545 then if condition becomes true since a box cannot fit in 15 px we have xGridPostions as 50 - 1 to accomodate the last box. 
+                                                                // The total usable canvas width now becomes 1530px;
+  } else {
+    xPositions = Math.floor(canvas.width / canvasElementsDim);
+  }
+  return xPositions;
+};
+// Number of Y positions possible at intervals of 30: (0, 30, 60, 90, ......)
+const yGridPositions = () => {
+  if (canvas.height / canvasElementsDim !== Math.floor(canvas.height / canvasElementsDim)) {   // Same concept as for xGridPostions.
+    yPositions = Math.floor(canvas.height / canvasElementsDim) - 1;
+  } else {
+    yPositions = Math.floor(canvas.height / canvasElementsDim);
+  }
+  return yPositions;
+};
 
 const generateSnake = () => {
-	const initialSnakeWidth = 30;
-	const initialSnakeHeight = 30;
-	const xPos = (window.innerWidth - initialSnakeWidth) / 2 ;
-	const yPos = (window.innerHeight - initialSnakeHeight) / 2;
+	const xPos = (canvas.width - canvasElementsDim) / 2 ;
+	const yPos = (canvas.height - canvasElementsDim) / 2;
 	ctx.fillStyle = "#FF0";
-	ctx.fillRect(xPos, yPos, initialSnakeWidth, initialSnakeHeight);
+	ctx.fillRect(xPos, yPos, canvasElementsDim, canvasElementsDim);
 	currentPos.snakeX = xPos;
 	currentPos.snakeY = yPos;
 }
 
 const generateApple = () => {
-	const appleWidth = 30;
-	const appleHeight = 30;
-	let randXPos, randYPos;
-	//Loop for checking if snake head exists at expected place for apple
-	//Realocate apple if true.
-	do {
-		randXPos = Math.floor(Math.random() * (window.innerWidth - appleWidth));
-		randYPos = Math.floor(Math.random() * (window.innerHeight - appleHeight));
-	}
-	while (randXPos===currentPos.snakeX && randYPos===currentPos.snakeY);
-
+	// Just track apple position outside this function and if condition matched then call the function again.
+  // don't check conditions here.
```

Comment:
It will still require a check, this is easier to interpret. Or go ahead to make a function check.

Reply:
I think we should merge this for now and work on a more elegant function.